lib/Interpreter/Transaction.cpp

```diff
@@ -67,11 +67,12 @@ namespace cling {
     for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
         if (DI->getKind() == Decl::LinkageSpec) {
-          for ( Decl *DI : LinkageSpecDecl::castToDeclContext(
-                static_cast<LinkageSpecDecl*>(DI))->decls()) {
-            if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
-              if (name.equals(ND->getNameAsString()))
-                return ND;
+          if (LinkageSpecDecl* LSD = cast<LinkageSpecDecl>(DI)) {
```

Comment:
But the first check is to avoid diving of into the type-system on every decl (which will likely be many).  I think it's a bit more efficient this way, but if you think otherwise I'll concede.